config/stp.py

```diff
@@ -0,0 +1,951 @@
+    for key in stp_vlan_keys:
+        if db.get_entry('STP_VLAN', key).get('enabled') == 'true':
+            count += 1
+    return count
+
+
+def vlan_enable_stp(db, vlan_name):
+    fvs = {'enabled': 'true',
+           'forward_delay': get_global_stp_forward_delay(db),
+           'hello_time': get_global_stp_hello_time(db),
+           'max_age': get_global_stp_max_age(db),
+           'priority': get_global_stp_priority(db)
+           }
+    if is_global_stp_enabled(db):
+        if get_stp_enabled_vlan_count(db) < get_max_stp_instances():
+            db.set_entry('STP_VLAN', vlan_name, fvs)
+        else:
+            logging.warning("Exceeded maximum STP configurable VLAN instances for {}".format(vlan_name))
+
+
+def interface_enable_stp(db, interface_name):
+    fvs = {'enabled': 'true',
+           'root_guard': 'false',
+           'bpdu_guard': 'false',
+           'bpdu_guard_do_disable': 'false',
+           'portfast': 'false',
+           'uplink_fast': 'false'
+           }
+    if is_global_stp_enabled(db):
+        db.set_entry('STP_PORT', interface_name, fvs)
+
+
+def is_vlan_configured_interface(db, interface_name):
+    intf_to_vlan_list = get_vlan_list_for_interface(db, interface_name)
+    if intf_to_vlan_list:  # if empty
+        return True
+    else:
+        return False
+
+
+def is_interface_vlan_member(db, vlan_name, interface_name):
+    ctx = click.get_current_context()
+    key = vlan_name + '|' + interface_name
+    entry = db.get_entry('VLAN_MEMBER', key)
+    if len(entry) == 0:  # if empty
+        ctx.fail("{} is not member of {}".format(interface_name, vlan_name))
+
+
+def get_vlan_list_for_interface(db, interface_name):
+    vlan_intf_info = db.get_table('VLAN_MEMBER')
+    vlan_list = []
+    for line in vlan_intf_info:
+        if interface_name == line[1]:
+            vlan_name = line[0]
+            vlan_list.append(vlan_name)
+    return vlan_list
+
+
+def get_pc_member_port_list(db):
+    pc_member_info = db.get_table('PORTCHANNEL_MEMBER')
+    pc_member_port_list = []
+    for line in pc_member_info:
+        intf_name = line[1]
+        pc_member_port_list.append(intf_name)
+    return pc_member_port_list
+
+
+def get_vlan_list_from_stp_vlan_intf_table(db, intf_name):
+    stp_vlan_intf_info = db.get_table('STP_VLAN_PORT')
+    vlan_list = []
+    for line in stp_vlan_intf_info:
+        if line[1] == intf_name:
+            vlan_list.append(line[0])
+    return vlan_list
+
+
+def get_intf_list_from_stp_vlan_intf_table(db, vlan_name):
+    stp_vlan_intf_info = db.get_table('STP_VLAN_PORT')
+    intf_list = []
+    for line in stp_vlan_intf_info:
+        if line[0] == vlan_name:
+            intf_list.append(line[1])
+    return intf_list
+
+
+def is_portchannel_member_port(db, interface_name):
+    pc_member_port_list = get_pc_member_port_list(db)
+    if interface_name in pc_member_port_list:
+        return True
+    else:
+        return False
```

Comment:
Would suggest simplify it without if else

return interface_name in pc_member_port_list

Reply:
Addressed